components/mpas-ocean/src/shared/mpas_ocn_vel_forcing_topographic_wave_drag.F

```diff
@@ -162,13 +181,100 @@ subroutine ocn_vel_forcing_topographic_wave_drag_init(err)!{{{
 
       integer, intent(out) :: err !< Output: error flag
 
+      real (kind=RKIND) :: topoDragCutoffDepth, topoDragCutoffWidth
+      real (kind=RKIND) :: omegaM2 = 1.405189e-4_RKIND
+      real (kind=RKIND) :: topoDragTurnOff
+
+      integer :: iEdge, k, kmin, kmax
+
       err = 0
 
+
       topographicWaveDragCoeff = 0.0_RKIND
 
       if (config_use_topographic_wave_drag) then
           topographicWaveDragOn = .true.
           topographicWaveDragCoeff = config_topographic_wave_drag_coeff
+          topoDragCutoffDepth = config_topographic_wave_drag_cutoff_depth
+          topoDragCutoffWidth = config_topographic_wave_drag_cutoff_width
+          if (config_topographic_wave_drag_scheme.EQ."JSL") then
+              call mpas_log_write("")
+              call mpas_log_write(" Topographic wave drag scheme is: Jayne and St. Laurent")
+              call mpas_log_write("")
+              tensorScheme = .false.
+
+              do iEdge = 1, nEdgesAll
+                  kmax =  maxLevelEdgeTop(iEdge)
+                  topographicWaveDrag(iEdge) = 0.0_RKIND
+                  if (kmax > 0) then
+                      k =  maxLevelEdgeTop(iEdge)
+                      topoDragTurnOff = (TANH((layerThickEdgeMean(k,iEdge)-topoDragCutoffDepth)/topoDragCutoffWidth)+1)/2
```

Comment:
I think I'm going to pull out the application `topoDragTurnOff` factor since it's common to all three drag schemes

Reply:
I agree, thanks!